oshdb-util/src/test/java/org/heigit/bigspatialdata/oshdb/util/time/IsoDateTimeParserTest.java

```diff
@@ -0,0 +1,175 @@
+package org.heigit.bigspatialdata.oshdb.util.time;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.Period;
+import java.util.Map;
+
+import static org.heigit.bigspatialdata.oshdb.util.time.IsoDateTimeParser.parseIsoDateTime;
+import static org.heigit.bigspatialdata.oshdb.util.time.IsoDateTimeParser.parseIsoPeriod;
+
+public class IsoDateTimeParserTest {
+
+    @Test
+    public void testParseISODateTime() throws Exception {
+        // test allowed variants
+
+        //Basic Dates
+        String[] yyyy = {"2020-01-01T00:00Z", "2020"};
+        String[] yyyymm = {"2020-02-01T00:00Z", "202002"};
+        String[] yyyymmdd = {"2020-02-17T00:00Z", "20200217"};
+
+        Assert.assertEquals(yyyy[0], parseIsoDateTime(yyyy[1]).toString());
+        Assert.assertEquals(yyyymm[0], parseIsoDateTime(yyyymm[1]).toString());
+        Assert.assertEquals(yyyymmdd[0], parseIsoDateTime(yyyymmdd[1]).toString());
+
+        //Extended Dates
+        String[] yyyy_mm = {"2020-02-01T00:00Z", "2020-02"};
+        String[] yyyy_mm_dd = {"2020-02-17T00:00Z", "2020-02-17"};
+
+        Assert.assertEquals(yyyy_mm[0], parseIsoDateTime(yyyy_mm[1]).toString());
+        Assert.assertEquals(yyyy_mm_dd[0], parseIsoDateTime(yyyy_mm_dd[1]).toString());
+
+        //Extended Date-Time
+        String[] yyyy_mm_dd_hh = {"2020-02-17T23:00Z", "2020-02-17T23"};
+        String[] yyyy_mm_dd_hhz = {"2020-02-17T23:00Z", "2020-02-17T23Z"};
+
+        String[] yyyy_mm_dd_hh_mm = {"2020-02-17T23:55Z", "2020-02-17T23:55"};
+        String[] yyyy_mm_dd_hh_mmz = {"2020-02-17T23:55Z", "2020-02-17T23:55Z"};
+
+        String[] yyyy_mm_dd_hh_mm_ss = {"2020-02-17T23:55:12Z", "2020-02-17T23:55:12"};
+        String[] yyyy_mm_dd_hh_mm_ssz = {"2020-02-17T23:55:12Z", "2020-02-17T23:55:12Z"};
+
+        String[] yyyy_mm_dd_hh_mm_ss_sss = {"2020-02-17T23:55:12.999Z", "2020-02-17T23:55:12.999"};
+        String[] yyyy_mm_dd_hh_mm_ss_sssz = {"2020-02-17T23:55:12.999Z", "2020-02-17T23:55:12.999Z"};
+
+        Assert.assertEquals(yyyy_mm_dd_hh[0], parseIsoDateTime(yyyy_mm_dd_hh[1]).toString());
+        Assert.assertEquals(yyyy_mm_dd_hhz[0], parseIsoDateTime(yyyy_mm_dd_hhz[1]).toString());
+
+        Assert.assertEquals(yyyy_mm_dd_hh_mm[0], parseIsoDateTime(yyyy_mm_dd_hh_mm[1]).toString());
+        Assert.assertEquals(yyyy_mm_dd_hh_mmz[0], parseIsoDateTime(yyyy_mm_dd_hh_mmz[1]).toString());
+
+        Assert.assertEquals(yyyy_mm_dd_hh_mm_ss[0], parseIsoDateTime(yyyy_mm_dd_hh_mm_ss[1]).toString());
+        Assert.assertEquals(yyyy_mm_dd_hh_mm_ssz[0], parseIsoDateTime(yyyy_mm_dd_hh_mm_ssz[1]).toString());
+
+        Assert.assertEquals(yyyy_mm_dd_hh_mm_ss_sss[0], parseIsoDateTime(yyyy_mm_dd_hh_mm_ss_sss[1]).toString());
+        Assert.assertEquals(yyyy_mm_dd_hh_mm_ss_sssz[0], parseIsoDateTime(yyyy_mm_dd_hh_mm_ss_sssz[1]).toString());
+
+    }
+
+
+    @Test(expected = Exception.class)
```

Comment:
Is a generic `Exception` sufficient here and in the following tests?

Reply:
made it more rigorous in 497792f28d14f7d992d9ba732362c6f23a545e00